pkg/source/kafka/kafka_source.go

```diff
@@ -259,14 +260,14 @@ func newKafkaSource(cfg *Configuration) (*kafkaSource, error) {
 		}
 	}
 
+	// returns nil, nil if provided empty certs
 	tlsConfig, err := common.CreateTLSConfiguration(cfg.CertFile, cfg.KeyFile, cfg.CaFile, cfg.SkipVerifyTLS)
 	if err != nil {
 		return nil, err
 	}
-	if tlsConfig != nil {
-		saramaConfig.Net.TLS.Config = tlsConfig
-		saramaConfig.Net.TLS.Enable = true
-	}
+
+	saramaConfig.Net.TLS.Enable = cfg.EnableTLS
+	saramaConfig.Net.TLS.Config = tlsConfig
```

Comment:
OK, so previously if TLS config was provided, TLS was enabled.

Now, providing only TLS config is not sufficient, you have to also explicitly TLS enable flag to `true` to make it work.  Correct?

Reply:
Correct. 

Because sometimes (eg. when connecting to EventHubs via the kafka protocol, or some SASL configurations), you need to enable TLS without providing certs.

Reply:
OK I understand no certs case.

With certs - I was thinking if would make sense to handle it like: if you provide TLS cert, TLS enable flag is automatically true? So you don't have to set it if you have a cert.

Reply:
I see the thinking but then you can have tls_enabled = false, but it still enables TLS - which is confusing.

We encountered this problem because I was trying to be too clever with the config - so my tendency here is to just keep it simple. In my head I ask "how would I document this configuration" - one way involves "if you set a then x and if you set b then if you provide c then y but if you set b and d then x". The other way is "set a to enable x, and provide b to configure x"

Happy for a dissenting view though

Reply:
Sounds good!